### DISCUSSION TITLE
How to access the OAuth access token?

### DISCUSSION THREAD

**Discussion by krisztiankocsis** [quarkusio/quarkus#33768] [Community]
Hi! We have a Quarkus app that is having OIDC based authentication to an external OIDC auth provider. How can we access the user's access_token from code, as we need it to perform authentication agains an external REST based service? Is it accessible via injected UserInfo?

**Comment by bulldog98** [comment]
There is a `Token` class you can inject it. Not sure if it's in classpath with your extensions

**Reply by sberyozkin** [reply]
@bulldog98 What `Token` class do you refer to ?

**Reply by bulldog98** [reply]
sorry it's called `Tokens` (io.quarkus.oidc.client.Tokens)

**Reply by sberyozkin** [reply]
@bulldog98 Np, that class is for the code depending on OIDC client, here the question is, how, after the authorization code flow has completed, to get the access token and use it to access some downstream service

**Comment by sberyozkin** [comment]
@krisztiankocsis Hi, 
https://quarkus.io/guides/security-openid-connect-client-reference#token-propagation-reactive

is the simplest option, the code flow token will be automatically propagated.

If you prefer you can inject it as either `JsonWebToken jwt` (if the code flow access token is JWT) or `AccessTokenCredential` and then manually pass it to the downstream service, but with the token propagation feature it is much easier

**Reply by krisztiankocsis** [reply]
Thanks you! I'll try the token propagation.
